x-pack/solutions/observability/plugins/synthetics/server/alert_rules/status_rule/status_rule_executor.ts

```diff
@@ -249,33 +264,108 @@ export class StatusRuleExecutor {
     return { from, to: 'now' };
   };
 
-  markDeletedConfigs(downConfigs: AlertStatusConfigs): Record<string, StaleDownConfig> {
+  markDeletedConfigs<T extends AlertStatusMetaData | AlertPendingStatusMetaData>(
+    configs: Record<string, T>
+  ): Record<string, T & StaleAlertMetadata> {
     const monitors = this.monitors;
-    const staleDownConfigs: AlertOverviewStatus['staleDownConfigs'] = {};
-    Object.keys(downConfigs).forEach((locPlusId) => {
-      const downConfig = downConfigs[locPlusId];
+    const staleConfigs: Record<string, T & StaleAlertMetadata> = {};
+
+    Object.keys(configs).forEach((locPlusId) => {
+      const config = configs[locPlusId];
       const monitor = monitors.find((m) => {
         return (
-          m.id === downConfig.configId ||
-          m.attributes[ConfigKey.MONITOR_QUERY_ID] === downConfig.monitorQueryId
+          m.id === config.configId ||
+          m.attributes[ConfigKey.MONITOR_QUERY_ID] === config.monitorQueryId
         );
       });
       if (!monitor) {
-        staleDownConfigs[locPlusId] = { ...downConfig, isDeleted: true };
-        delete downConfigs[locPlusId];
+        staleConfigs[locPlusId] = { ...config, isDeleted: true };
+        delete configs[locPlusId];
       } else {
         const { locations } = monitor.attributes;
-        const isLocationRemoved = !locations.some((l) => l.id === downConfig.locationId);
+        const isLocationRemoved = !locations.some((l) => l.id === config.locationId);
         if (isLocationRemoved) {
-          staleDownConfigs[locPlusId] = { ...downConfig, isLocationRemoved: true };
-          delete downConfigs[locPlusId];
+          staleConfigs[locPlusId] = { ...config, isLocationRemoved: true };
+          delete configs[locPlusId];
         }
       }
     });
 
-    return staleDownConfigs;
+    return staleConfigs;
+  }
+
+  schedulePendingAlertPerConfigIdPerLocation({
```

Comment:
need to add unit tests for this function

Reply:
Done